File: C programs/Quick_sort.c
#include<stdlib.h>
#include<stdio.h>
#include<string.h>
#define maxi 10
int max;
void swap(char*arr,int p1,int p2)
{
    char temp[maxi]=arr[p1];
    arr[p1]=arr[p2];
    arr[p2]=temp[maxi];
}
int partition(char*arr,int l,int h)
{
    int pi=h;
    int i=l-1;
    for(int j=0;j<h;j++){
        if(arr[pi]>arr[j]){
                i++;
        swap(arr,i,j);
        }
    }
    swap(arr,i+1,h);
    return(i+1);
}
void Sort(char*arr,int l,int h)
{
    if(l<h){
    int pi=partition(arr,l,h);

    Sort(arr,l,pi-1);
    Sort(arr,pi+1,h);
    }
}
void display(char*arr)
{
    printf("The Array is : \n");
    for(int i=0;i<strlen(arr);i++)
    {
        printf("  %c ",arr[i]);
    }
    printf("\n");
}
void main()
{
    char*arr;
    printf("Enter the Size of the Array :\n");
    scanf("%d",&max);
    arr=(char*)malloc(max*sizeof(char));
    for(int i=0;i<max;i++)
    {
        printf("Enter the Item %d ",i);
        scanf("%s",&arr[i]);
    }
    printf("\n\n");
    display(arr);
    printf("The Sorted Array is \n");
    Sort(arr,0,max-1);
    display(arr);
}

File: JAVA programs/Quick_Sort.java
import java.util.Scanner;

public class Quick_Sort {
    public static void main(String[]args)
    {
        Scanner S=new Scanner(System.in);
        int n;
        System.out.println("Enter the Length of the Array :");
        n=S.nextInt();
        int[]arr=new int[n];
        System.out.println("Enter the Array Elements : ");
        for(int i=0;i<n;i++)
        {
            System.out.print("Element "+(i+1)+" : ");
            arr[i]=S.nextInt();
        }
        System.out.println("The Array is : ");
        display(arr);
        System.out.println("The Sorted Array Using Quick Sort is : ");
        QuickSort(arr,0, arr.length-1);
        display(arr);
    }
   static void display(int[]arr)
   {
       for(int i=0;i<arr.length;i++)
       {
           System.out.print("\t"+arr[i]);
       }
   }
   static void QuickSort(int[]arr,int l,int u)
   {
       if(l<u)
       {
           int pa=Part(arr,l,u);
           QuickSort(arr,l,pa-1);
           QuickSort(arr,pa+1,u);
       }
   }
   static void Swap(int[]arr,int i,int j)
   {
       int temp=arr[i];
       arr[i]=arr[j];
       arr[j]=temp;
   }
   static int Part(int[]arr,int l,int u)
   {
       int pivot=arr[u];
       int k=l-1;
       for(int i=l;i<u-1;i++)
       {
           if(pivot>arr[i])
           {
               k++;
               Swap(arr,k,i);
           }
           Swap(arr,k,(i+1));
           return(i+1);
       }
       return(-1);
   }
}
